src/wwwroot/template-editor/adapterService.js

```diff
@@ -0,0 +1,356 @@
+         * }
+         */
+        console.log("getPreviewImage");
+        var response = {
+              ThumbnailUrl: 'https://i.vimeocdn.com/video/699535210.webp', // jshint ignore:line
+              Title: 'Test video',
+              ProviderName: 'Youtube',
+              //If Add reproductor is checked below property is added
+              ThumbnailUrlWithPlay: 'https://i.vimeocdn.com/filter/overlay?src=' + 'https://i.vimeocdn.com/video/699535210.webp' + '&src=https://integrationstore-b0c3f53658fe7a75.microservice.createsend.com/files/9392B9D9-F380-42FC-9571-7E109B7A1C26/youtube-play-button-overlay.png'
+            };
+        var fakePromise = $q.defer();
+        fakePromise.resolve(response);
+        return fakePromise.promise;
+      }
+
+      function getImagesForCampaign(start, howMany, query, sortingCriteria) {
+        /**
+         * Return a preview images to image gallery
+         * @method
+         * @argument {integer} start - Initial position index 
+         * @argument {integer} howMany - Offset position
+         * @argument {object} query - Query model, Example:
+         * {
+         *   query: ''
+         * }
+         * @argument {string} sortingCriteria - Image sorting criteria
+         */
+        console.log("getImagesForCampaign");
+        var images = [{
+          name: 'fakeImage1.jpg',
+          lastModifiedDate: 'Thu Oct 19 2017 00:35:33 GMT+0100 (GMT Summer Time)',
+          size: 1024,
+          type: '.jpg',
+          url: 'http://cdn.fromdoppler.com/mseditor/v1.0.0-build306/uploads/fakeImage1.jpg',
+          thumbnailUrl: 'http://cdn.fromdoppler.com/mseditor/v1.0.0-build306/uploads/fakeImage1.jpg',
+          thumbnailUrl150: 'http://cdn.fromdoppler.com/mseditor/v1.0.0-build306/uploads/fakeImage1.jpg'
+        }];
+        var response = {
+          data:{
+            images: images,
+            count: 1
+          }
+        };
+        var fakePromise = $q.defer();
+        fakePromise.resolve(response);
+        return fakePromise.promise;
+      }
+
+      function deleteImages(filenames) {
+        /**
+         * Return an array of promises to delete individual images
+         * @method
+         * @argument {array} params - Example
+         * [
+         *   {
+         *     name: 'image1.jpg',
+         *     lastModifiedDate: '2008-04-05 07:10:14.000000000 +0100',
+         *     size: 1024,
+         *     type: '.jpg',
+         *     url: 'uploads/image1.jpg',
+         *     thumbnailUrl: 'uploads/image1.jpg',
+         *     thumbnailUrl150: 'uploads/image1.jpg'
+         *   },
+         *   {
+         *     name: 'image2.jpg',
+         *     lastModifiedDate: '2008-02-05 01:10:14.000000000 +0100',
+         *     size: 332,
+         *     type: '.jpg',
+         *     url: 'uploads/image2.jpg',
+         *     thumbnailUrl: 'uploads/image2.jpg',
+         *     thumbnailUrl150: 'uploads/image2.jpg'
+         *   }
+         * ]
```

Comment:
I researched a little and I did not found a clear way to include long code examples in arguments or descriptions. But it seems that [_Description_ (at least) supports Markdown format](http://usejsdoc.org/tags-description.html#overview), maybe you could add indentation or ```symbols to represent the code as Markdown does.

Another option could be use _typedef_, but I think that it will take more time.

Reply:
Ohhh, it looks like the best option: http://usejsdoc.org/tags-param.html#parameters-with-properties

```js
/**
 * Assign the project to an employee.
 * @param {Object} employee - The employee who is responsible for the project.
 * @param {string} employee.name - The name of the employee.
 * @param {string} employee.department - The employee's department.
 */
Project.prototype.assign = function(employee) {
    // ...
};

```

But you should describe a schema in place of simply put an example.